File: game-engine/Domain/Models/DTOs/GameStateDto.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Domain.Models.DTOs
{
    public class GameStateDto
    {
        public World World { get; set; }
        public List<BotDto> Bots { get; set; }
        public Guid BotId { get; set; }
        public List<PopulationTier> PopulationTiers { get; set; }

        public override string ToString()
        {
            return JsonConvert.SerializeObject(this);
        }
    }
}

File: starter-bots/KotlinBot/src/main/kotlin/arctica/service/model/dto/GameStateDto.kt
package arctica.service.model.dto

import BotDto
import arctica.service.model.PopulationTier
import arctica.service.model.World
import java.util.*

data class GameStateDto(
    val world: World,
    val bots: List<BotDto>,
    val botId: UUID,
    val populationTiers: List<PopulationTier>
)
